extensions/picocli/deployment/src/main/java/io/quarkus/picocli/deployment/PicocliProcessor.java

```diff
@@ -99,6 +101,17 @@ void picocliRunner(ApplicationIndexBuildItem applicationIndex,
             additionalBean.produce(AdditionalBeanBuildItem.unremovableOf(DefaultPicocliCommandLineFactory.class));
             quarkusApplicationClass.produce(new QuarkusApplicationClassBuildItem(PicocliRunner.class));
         }
+
+        // Make all classes that can be instantiated by IFactory unremovable
+        unremovableBean.produce(UnremovableBeanBuildItem.beanTypes(CommandLine.ITypeConverter.class,
```

Comment:
Ok - so this is doing it for all instances even if not referenced by picocli config.

Just curious why you do it this way when we only have to do it for those pointed to by users annotation?

Ease of coding in quarkus or that you don't believe it will matter enough in extra memory used ?

Reply:
> Ok - so this is doing it for all instances even if not referenced by picocli config.

All it does it make beans of this type unremovable - nothing more

Reply:
> Just curious why you do it this way when we only have to do it for those pointed to by users annotation?

This way is much much cleaner and also makes chances of copy-paste errors in the code far lower.